src/components/preview/preview.component.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { MarginProps } from "styled-system";
+
+import PreviewPlaceholder, {
+  PreviewPlaceholderProps,
+} from "./__internal__/preview-placeholder.component";
+import { StyledPreview } from "./preview.style";
+import { filterStyledSystemMarginProps } from "../../style/utils";
+
+export interface PreviewProps
+  extends Partial<Omit<PreviewPlaceholderProps, "index">>,
+    MarginProps {
+  /** Children content to render in the component. */
+  children?: React.ReactNode;
+  /* Provides more control over when in a loading state. */
+  loading?: boolean;
+}
+
+export const Preview = ({
+  children,
+  loading,
+  lines = 1,
+  ...props
+}: PreviewProps) => {
+  const marginProps = filterStyledSystemMarginProps(props);
+  const hasPlaceholder = typeof loading !== "undefined" ? loading : !children;
```

Comment:
```suggestion
  const hasPlaceholder = loading === undefined ? !children : loading;
```

Not very significant but I prefer to write it this way, for 2 reasons:
1) `type of loading !== "undefined"` is exactly the same as `loading !== undefined` (the only difference is if `loading` isn't even declared, in which case my version would crash whereas yours would give `false` - but the `loading` is defined in the props argument here so that's never going to be an issue)
2) I much prefer ternaries and if/else to be the "right" way round - check if two things are equal, not if they're not equal. Mostly because I'm stupid :-)

This is just a suggestion, if you prefer it your way then you don't need to justify it to me, just leave it!